File: app/src/main/java/com/pixel8/dynamicisland/MainActivity.java
package com.pixel8.dynamicisland;

import android.app.Activity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.Color;
import android.graphics.Typeface;
import android.graphics.drawable.GradientDrawable;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.provider.Settings;
import android.util.TypedValue;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.GridLayout;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.SeekBar;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends Activity {

    private SharedPreferences prefs;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        prefs = getSharedPreferences("island_prefs", MODE_PRIVATE);

        ScrollView scrollView = new ScrollView(this);
        scrollView.setBackgroundColor(Color.parseColor("#090A0F"));

        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setPadding(dpToPx(20), dpToPx(36), dpToPx(20), dpToPx(36));
        root.setGravity(Gravity.CENTER_HORIZONTAL);
        scrollView.addView(root);

        // Header
        TextView title = new TextView(this);
        title.setText("Pixel 8 Dynamic Island");
        title.setTextColor(Color.WHITE);
        title.setTextSize(TypedValue.COMPLEX_UNIT_SP, 22);
        title.setTypeface(Typeface.DEFAULT_BOLD);
        root.addView(title);

        TextView subtitle = new TextView(this);
        subtitle.setText("إعدادات وتخصيص حجم وموضع الجزيرة 🏝️");
        subtitle.setTextColor(Color.parseColor("#38BDF8"));
        subtitle.setTextSize(TypedValue.COMPLEX_UNIT_SP, 13);
        LinearLayout.LayoutParams subP = new LinearLayout.LayoutParams(-2, -2);
        subP.setMargins(0, dpToPx(4), 0, dpToPx(20));
        subtitle.setLayoutParams(subP);
        root.addView(subtitle);

        // Section: Main Switch
        root.addView(createSectionTitle("الحالة والتشغيل"));
        root.addView(createButton("⚡ إعادة تشغيل / تفعيل الجزيرة", "#0284C7", new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                    if (!Settings.canDrawOverlays(MainActivity.this)) {
                        startActivity(new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
                                Uri.parse("package:" + getPackageName())));
                    } else {
                        startService(new Intent(MainActivity.this, DynamicIslandService.class));
                        Toast.makeText(MainActivity.this, "تم تفعيل الجزيرة بالحجم المخصص!", Toast.LENGTH_SHORT).show();
                    }
                }
            }
        }));

        // Section: Calibration Sliders (تخصيص الحجم والموضع)
        root.addView(createSectionTitle("تخصيص موضع وأبعاد الكاميرا (Pixel 8)"));

        // 1. عرض الجزيرة (Width)
        final TextView tvW = new TextView(this);
        int curW = prefs.getInt("compact_width", 125);
        tvW.setText("عرض الجزيرة: " + curW + " dp");
        tvW.setTextColor(Color.LTGRAY);
        root.addView(tvW);

        SeekBar sbW = createSeekBar(80, 260, curW, new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tvW.setText("عرض الجزيرة: " + progress + " dp");
                prefs.edit().putInt("compact_width", progress).apply();
                sendBroadcast(new Intent("com.pixel8.dynamicisland.UPDATE_SIZE"));
            }
            @Override public void onStartTrackingTouch(SeekBar seekBar) {}
            @Override public void onStopTrackingTouch(SeekBar seekBar) {}
        });
        root.addView(sbW);

        // 2. المسافة من الأعلى (Y Position)
        final TextView tvY = new TextView(this);
        int curY = prefs.getInt("notch_y", 8);
        tvY.setText("المسافة من أعلى الشاشة: " + curY + " dp");
        tvY.setTextColor(Color.LTGRAY);
        root.addView(tvY);

        SeekBar sbY = createSeekBar(0, 40, curY, new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tvY.setText("المسافة من أعلى الشاشة: " + progress + " dp");
                prefs.edit().putInt("notch_y", progress).apply();
                sendBroadcast(new Intent("com.pixel8.dynamicisland.UPDATE_SIZE"));
            }
            @Override public void onStartTrackingTouch(SeekBar seekBar) {}
            @Override public void onStopTrackingTouch(SeekBar seekBar) {}
        });
        root.addView(sbY);

        // Section: Live Activities Grid
        root.addView(createSectionTitle("اختبار الأنشطة التفاعلية (Live Activities)"));

        GridLayout grid = new GridLayout(this);
        grid.setColumnCount(2);
        grid.setLayoutParams(new LinearLayout.LayoutParams(-1, -2));

        addGridButton(grid, "🎵 الموسيقى والصوت", "Starboy - The Weeknd", "music", "🎵");
        addGridButton(grid, "⏱️ مؤقت 05:00 دقيقة", "متبقي 04:59 ثانية", "timer", "⏱️");
        addGridButton(grid, "💬 إشعار واتساب", "رسالة جديدة: مرحباً بك!", "whatsapp", "🟢");
        addGridButton(grid, "⚡ الشحن السريع 30W", "البطارية 88%", "charging", "⚡");
        addGridButton(grid, "📞 مكالمة هاتفية", "مكالمة واردة...", "call", "📞");
        addGridButton(grid, "🎧 سماعات Buds", "متصلة 100%", "buds", "🎧");

        root.addView(grid);

        setContentView(scrollView);
    }

    private SeekBar createSeekBar(int min, int max, int current, SeekBar.OnSeekBarChangeListener listener) {
        SeekBar sb = new SeekBar(this);
        sb.setMax(max - min);
        sb.setProgress(current - min);
        LinearLayout.LayoutParams p = new LinearLayout.LayoutParams(-1, -2);
        p.setMargins(0, dpToPx(4), 0, dpToPx(14));
        sb.setLayoutParams(p);
        sb.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                listener.onProgressChanged(seekBar, progress + min, fromUser);
            }
            @Override public void onStartTrackingTouch(SeekBar seekBar) {}
            @Override public void onStopTrackingTouch(SeekBar seekBar) {}
        });
        return sb;
    }

    private void addGridButton(GridLayout grid, final String title, final String text, final String type, final String icon) {
        Button btn = new Button(this);
        btn.setText(title);
        btn.setTextColor(Color.WHITE);
        btn.setTextSize(TypedValue.COMPLEX_UNIT_SP, 12);
        btn.setTypeface(Typeface.DEFAULT_BOLD);
        btn.setBackground(createCurvedBg(dpToPx(12), Color.parseColor("#171923"), Color.parseColor("#2D3748")));

        GridLayout.LayoutParams p = new GridLayout.LayoutParams();
        p.width = 0;
        p.height = dpToPx(60);
        p.columnSpec = GridLayout.spec(GridLayout.UNDEFINED, 1, 1f);
        p.setMargins(dpToPx(4), dpToPx(4), dpToPx(4), dpToPx(4));
        btn.setLayoutParams(p);

        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startService(new Intent(MainActivity.this, DynamicIslandService.class));
                Intent intent = new Intent("com.pixel8.dynamicisland.NOTIF");
                intent.putExtra("title", title);
                intent.putExtra("text", text);
                intent.putExtra("type", type);
                intent.putExtra("icon", icon);
                sendBroadcast(intent);
            }
        });

        grid.addView(btn);
    }

    private TextView createSectionTitle(String text) {
        TextView tv = new TextView(this);
        tv.setText(text);
        tv.setTextColor(Color.parseColor("#94A3B8"));
        tv.setTextSize(TypedValue.COMPLEX_UNIT_SP, 13);
        tv.setTypeface(Typeface.DEFAULT_BOLD);
        LinearLayout.LayoutParams p = new LinearLayout.LayoutParams(-1, -2);
        p.setMargins(0, dpToPx(14), 0, dpToPx(8));
        tv.setLayoutParams(p);
        return tv;
    }

    private Button createButton(String text, String colorHex, View.OnClickListener listener) {
        Button btn = new Button(this);
        btn.setText(text);
        btn.setTextColor(Color.WHITE);
        btn.setTextSize(TypedValue.COMPLEX_UNIT_SP, 13);
        btn.setTypeface(Typeface.DEFAULT_BOLD);
        btn.setBackground(createCurvedBg(dpToPx(12), Color.parseColor(colorHex), Color.TRANSPARENT));
        btn.setOnClickListener(listener);

        LinearLayout.LayoutParams p = new LinearLayout.LayoutParams(-1, dpToPx(50));
        p.setMargins(0, 0, 0, dpToPx(12));
        btn.setLayoutParams(p);
        return btn;
    }

    private GradientDrawable createCurvedBg(int radiusPx, int bgColor, int strokeColor) {
        GradientDrawable shape = new GradientDrawable();
        shape.setCornerRadius(radiusPx);
        shape.setColor(bgColor);
        if (strokeColor != Color.TRANSPARENT) shape.setStroke(dpToPx(1), strokeColor);
        return shape;
    }

    private int dpToPx(int dp) {
        return (int) TypedValue.applyDimension(
                TypedValue.COMPLEX_UNIT_DIP, dp, getResources().getDisplayMetrics());
    }
}

File: app/src/main/java/com/pixel8/dynamicisland/MainActivity.kt
package com.pixel8.dynamicisland

import android.content.Intent
import android.content.SharedPreferences
import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.os.Bundle
import android.provider.Settings
import android.view.Gravity
import android.widget.Button
import android.widget.LinearLayout
import android.widget.ScrollView
import android.widget.SeekBar
import android.widget.Switch
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    companion object {
        const val ACTION_UPDATE_CONFIG = "com.pixel8.dynamicisland.UPDATE_CONFIG"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val prefs = getSharedPreferences("island_prefs", MODE_PRIVATE)

        val scrollView = ScrollView(this).apply {
            setBackgroundColor(Color.parseColor("#0F0F0F"))
        }

        val root = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(40, 50, 40, 50)
        }

        // ترويسة أنيقة
        val title = TextView(this).apply {
            text = "Pixel 8 Dynamic Island"
            setTextColor(Color.WHITE)
            textSize = 22f
            typeface = android.graphics.Typeface.DEFAULT_BOLD
        }
        root.addView(title)

        val subTitle = TextView(this).apply {
            text = "تحكم شامل: الشاحن • الموسيقى • الإشعارات"
            setTextColor(Color.parseColor("#888888"))
            textSize = 12f
            setPadding(0, 4, 0, 25)
        }
        root.addView(subTitle)

        // 1. بطاقة الأذونات
        val permCard = createCard().apply {
            addView(createCardTitle("🛡️ أذونات التشغيل"))
            addView(createCleanButton("1. إذن الظهور فوق الشاشة (Accessibility)", "#1E88E5").apply {
                setOnClickListener { startActivity(Intent(Settings.ACTION_ACCESSIBILITY_SETTINGS)) }
            })
            addView(createCleanButton("2. إذن قراءة الإشعارات والموسيقى", "#2E7D32").apply {
                setOnClickListener { startActivity(Intent(Settings.ACTION_NOTIFICATION_LISTENER_SETTINGS)) }
            })
        }
        root.addView(permCard)

        // 2. بطاقة الشاحن والبطارية
        val batteryCard = createCard().apply {
            addView(createCardTitle("⚡ إعدادات الشاحن والبطارية"))
            addView(createSwitchRow("أنيميشن توصيل الشاحن السريع", "enable_charging", true, prefs))
            addView(createSwitchRow("تنبيه اكتمال الشحن (100%)", "enable_full_battery", true, prefs))
            addView(createSwitchRow("تنبيه انخفاض البطارية (20%)", "enable_low_battery", true, prefs))
        }
        root.addView(batteryCard)

        // 3. بطاقة الموسيقى
        val musicCard = createCard().apply {
            addView(createCardTitle("🎵 إعدادات مشغل الموسيقى"))
            addView(createSwitchRow("إظهار شريط الموسيقى عند التشغيل", "enable_music", true, prefs))
            addView(createSwitchRow("موجة صوتية متحركة (Waveform)", "enable_waveform", true, prefs))
            addView(createSwitchRow("عرض تفاصيل الأغنية والفنان", "enable_track_info", true, prefs))
        }
        root.addView(musicCard)

        // 4. بطاقة محاذاة الكاميرا
        val positionCard = createCard().apply {
            addView(createCardTitle("📐 أبعاد وموضع الكاميرا"))

            val currentY = prefs.getInt("island_y", 12)
            val txtY = TextView(context).apply {
                text = "المسافة من الأعلى: $currentY px"
                setTextColor(Color.parseColor("#AAAAAA"))
                textSize = 12f
                setPadding(0, 5, 0, 5)
            }
            addView(txtY)

            val seekY = SeekBar(context).apply {
                max = 60
                progress = currentY
                setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
                    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
                        txtY.text = "المسافة من الأعلى: $progress px"
                        prefs.edit().putInt("island_y", progress).apply()
                        sendLiveUpdate(progress, prefs.getInt("island_width", 280))
                    }
                    override fun onStartTrackingTouch(seekBar: SeekBar?) {}
                    override fun onStopTrackingTouch(seekBar: SeekBar?) {}
                })
            }
            addView(seekY)

            val currentW = prefs.getInt("island_width", 280)
            val txtW = TextView(context).apply {
                text = "عرض الكبسولة في وضع السكون: $currentW px"
                setTextColor(Color.parseColor("#AAAAAA"))
                textSize = 12f
                setPadding(0, 15, 0, 5)
            }
            addView(txtW)

            val seekW = SeekBar(context).apply {
                min = 180
                max = 380
                progress = currentW
                setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
                    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
                        txtW.text = "عرض الكبسولة في وضع السكون: $progress px"
                        prefs.edit().putInt("island_width", progress).apply()
                        sendLiveUpdate(prefs.getInt("island_y", 12), progress)
                    }
                    override fun onStartTrackingTouch(seekBar: SeekBar?) {}
                    override fun onStopTrackingTouch(seekBar: SeekBar?) {}
                })
            }
            addView(seekW)
        }
        root.addView(positionCard)

        // زر التجربة
        val btnTest = createCleanButton("✨ تجربة إشعار تفاعلي", "#242426").apply {
            setTextColor(Color.parseColor("#00E676"))
            setOnClickListener {
                val intent = Intent(NotificationService.ACTION_NEW_NOTIFICATION).apply {
                    putExtra("title", "رسالة جديدة")
                    putExtra("text", "إشعارات الجزيرة تعمل بامتياز على Pixel 8 🔥")
                }
                sendBroadcast(intent)
                Toast.makeText(this@MainActivity, "تم إرسال إشعار تجريبي!", Toast.LENGTH_SHORT).show()
            }
        }
        root.addView(btnTest)

        scrollView.addView(root)
        setContentView(scrollView)
    }

    private fun createCard(): LinearLayout {
        return LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(35, 28, 35, 28)
            background = GradientDrawable().apply {
                setColor(Color.parseColor("#1C1C1E"))
                cornerRadius = 26f
            }
            layoutParams = LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.WRAP_CONTENT
            ).apply {
                setMargins(0, 0, 0, 20)
            }
        }
    }

    private fun createCardTitle(text: String): TextView {
        return TextView(this).apply {
            this.text = text
            setTextColor(Color.WHITE)
            textSize = 14f
            typeface = android.graphics.Typeface.DEFAULT_BOLD
            setPadding(0, 0, 0, 12)
        }
    }

    private fun createSwitchRow(title: String, prefKey: String, defVal: Boolean, prefs: SharedPreferences): LinearLayout {
        return LinearLayout(this).apply {
            orientation = LinearLayout.HORIZONTAL
            gravity = Gravity.CENTER_VERTICAL
            setPadding(0, 8, 0, 8)

            val label = TextView(context).apply {
                text = title
                setTextColor(Color.parseColor("#CCCCCC"))
                textSize = 13f
                layoutParams = LinearLayout.LayoutParams(0, LinearLayout.LayoutParams.WRAP_CONTENT, 1f)
            }

            val sw = Switch(context).apply {
                isChecked = prefs.getBoolean(prefKey, defVal)
                setOnCheckedChangeListener { _, isChecked ->
                    prefs.edit().putBoolean(prefKey, isChecked).apply()
                }
            }

            addView(label)
            addView(sw)
        }
    }

    private fun createCleanButton(text: String, colorHex: String): Button {
        return Button(this).apply {
            this.text = text
            setTextColor(Color.WHITE)
            textSize = 12f
            background = GradientDrawable().apply {
                setColor(Color.parseColor(colorHex))
                cornerRadius = 18f
            }
            layoutParams = LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT, 115
            ).apply {
                setMargins(0, 4, 0, 8)
            }
        }
    }

    private fun sendLiveUpdate(y: Int, width: Int) {
        val intent = Intent(ACTION_UPDATE_CONFIG).apply {
            putExtra("island_y", y)
            putExtra("island_width", width)
        }
        sendBroadcast(intent)
    }
}
